src/java.base/share/classes/jdk/internal/foreign/AbstractMemorySegmentImpl.java

```diff
@@ -394,18 +389,13 @@ public RuntimeException apply(String s, List<Number> numbers) {
     }
 
     @Override
-    public Scope scope() {
+    public MemorySessionImpl scope() {
```

Comment:
Should this be `final` and `@ForceInline` to better match the old behavior of `sessionImpl`?